pennylane_lightning/lightning_tensor/quimb/_mps.py

```diff
@@ -0,0 +1,428 @@
+        * Supports any three or four-qubit operations that provide a decomposition method.
+        * Currently does not support finite shots.
+        """
+
+        program = TransformProgram()
+
+        program.add_transform(validate_measurements, name=self.name_interf)
+        program.add_transform(validate_observables, accepted_observables, name=self.name_interf)
+        program.add_transform(validate_device_wires, self._wires, name=self.name_interf)
+        program.add_transform(
+            decompose,
+            stopping_condition=stopping_condition,
+            skip_initial_state_prep=True,
+            name=self.name_interf,
+        )
+        program.add_transform(qml.transforms.broadcast_expand)
+
+        return program
+
+    # pylint: disable=unused-argument
+    def execute(
+        self,
+        circuits: QuantumTape_or_Batch,
+        execution_config: ExecutionConfig = DefaultExecutionConfig,
+    ) -> Result_or_ResultBatch:
+        """Execute a circuit or a batch of circuits and turn it into results.
+
+        Args:
+            circuits (Union[QuantumTape, Sequence[QuantumTape]]): the quantum circuits to be executed
+            execution_config (ExecutionConfig): a datastructure with additional information required for execution
+
+        Returns:
+            TensorLike, tuple[TensorLike], tuple[tuple[TensorLike]]: A numeric result of the computation.
+        """
+
+        results = []
+        for circuit in circuits:
+            circuit = circuit.map_to_standard_wires()
+            results.append(self.simulate(circuit))
+
+        return tuple(results)
+
+    def simulate(self, circuit: QuantumScript) -> Result:
+        """Simulate a single quantum script. This function assumes that all operations provide matrices.
+
+        Args:
+            circuit (QuantumScript): The single circuit to simulate.
+
+        Returns:
+            Tuple[TensorLike]: The results of the simulation.
+        """
+
+        self._reset_state()
+
+        for op in circuit.operations:
+            self._apply_operation(op)
+
+        if not circuit.shots:
+            if len(circuit.measurements) == 1:
+                return self.measurement(circuit.measurements[0])
+            return tuple(self.measurement(mp) for mp in circuit.measurements)
+
+        raise NotImplementedError
+
+    def _apply_operation(self, op: qml.operation.Operator) -> None:
+        """Apply a single operator to the circuit, keeping the state always in a MPS form.
+
+        Internally it uses `quimb`'s `apply_gate` method. For operations that act on more than two wires,
+        it decomposes them first into operations that act on at most two wires.
+
+        Args:
+            op (Operator): The operation to apply.
+        """
+
+        if len(op.wires) <= 2:
+            self._circuitMPS.apply_gate(op.matrix(), *op.wires, **self._gate_opts)
+        else:
+            decom_ops = decompose_recursive(op)
```

Comment:
Should we move this in `lightning_tensor`'s `preprocess` as done in [LQubit](https://github.com/PennyLaneAI/pennylane-lightning/blob/master/pennylane_lightning/lightning_qubit/lightning_qubit.py#L536)?

Reply:
`execute` is meant to execute only tapes with natively supported operations. If that's only `RX`, `RY`, `RZ` and `CNOT`,  so be it. That's all we need to test before moving on to fixing `setup.py` and improving `preprocessing`.

Reply:
As discussed on Slack, we'll remove such decomposition once `quimb 1.8.1` is available, so I'll mark this as resolved for now (I created a story in the second epic so we don't forget)